astore/server/auth/factory.go

```diff
@@ -117,7 +117,9 @@ func WithCA(fileContent []byte) Modifier {
 func WithPrincipals(raw string) Modifier {
 	return func(server *Server) error {
 		splitString := strings.Split(raw, ",")
-		server.principals = splitString
+		if len(splitString) >= 1 && splitString[0] != "" {
+			server.principals = splitString
```

Comment:
Good catch! The problem is that empty string will lead to a split of 1, with an empty string, correct?

It took me a couple seconds to be sure your code above was correct. How about this:
```
raw = strings.TrimSpace(raw)
if raw != "" {
  server.principals = strings.Split(raw, ",")
}
```

Reply:
Ah I'm going to just merge this as is because I'm going to PR kcobra to have the VarStringArray like standard cobra